presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/AlternativeNullableLongState.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.state;
+
+import com.facebook.presto.spi.function.AccumulatorStateMetadata;
+
+@AccumulatorStateMetadata(stateSerializerClass = AlternativeNullableLongStateSerializer.class)
+public interface AlternativeNullableLongState
```

Comment:
I think this and the serializer should have a more descriptive name, as Alternative doesn't tell you much about what it's for. don't have ideas right now.  Also, add a comment to the top of this class to explaining its purpose.